definitions/checks/foreman/check_checkpoint_segments.rb

```diff
@@ -0,0 +1,50 @@
+module Checks
+  module Foreman
+    class CheckpointSegments < ForemanMaintain::Check
+      metadata do
+        label :check_checkpoint_segments
+        description 'Check if checkpoint_segments parameter persent in custom hiera'
+        tags :pre_upgrade
+      end
+
+      HIERA_FILE = '/etc/foreman-installer/custom-hiera.yaml'.freeze
+
+      def run
+        files = []
+        files << feature(:foreman_database).config_files[0] if check_postgres_config
+        files << HIERA_FILE if check_custom_hiera
+        unless files.empty?
+          message = ERB.new(<<-BLOCK.strip_heredoc).result(binding)
+          ERROR:
+            checkpoint_segments tuning found.
+            This tuning option is no longer valid in PostgreSQL 9.5+
+            Please remove this from the following locations and then re-run the foreman-maintain upgrade check:
+            <% files.each do |file| %>
+              - <%= file %>
+            <% end %>
+          BLOCK
```

Comment:
```suggestion
failure_message = <<-MESSAGE.strip_heredoc

ERROR: Tuning option 'checkpoint_segments' found.
This option is no longer valid for PostgreSQL 9.5 and onwards.
Please remove it from following files and re-run the command.
#{files_to_modify.join("\n")}
MESSAGE
```

To maintain consistency could you make use of `MESSAGE.strip_heredoc`.